lib/instance.js

```diff
@@ -236,15 +236,14 @@ module.exports = (function() {
 
         if (options.attributes) {
           keys = options.attributes;
-          if (this.Model._hasVirtualAttributes) {
-            keys = keys.concat(this.Model._virtualAttributes);
-          }
 
           if (this.options.includeNames) {
```

Comment:
This is not true right now. If you have a look at the statements you'll realize that if keys are set by virtual attributes they would be overridden by the next statement. I handle it exactly the same as it was before, just with one if statement less.